tooling/nargo_cli/src/cli/compile_cmd.rs

```diff
@@ -61,7 +61,12 @@ pub(crate) fn run(
     let default_selection =
         if args.workspace { PackageSelection::All } else { PackageSelection::DefaultOrAll };
     let selection = args.package.map_or(default_selection, PackageSelection::Selected);
-    let workspace = resolve_workspace_from_toml(&toml_path, selection)?;
+
+    let workspace = resolve_workspace_from_toml(
+        &toml_path,
+        selection,
+        Some(NOIR_ARTIFACT_VERSION_STRING.to_owned()),
```

Comment:
The check is not skipped for compile however, so you can test this by trying to compile any of our execution_success tests. As of this comment, these will be failing because their Nargo.toml files explicitly say "0.1". This is also why this PR is a breaking change; our init_cmd explicitly specifies the compiler_version, so this will likely break quite a lot of projects.